src/System.Memory/src/System/Buffers/ReadOnlySequence.cs

```diff
@@ -363,7 +363,7 @@ public ReadOnlySequence<T> Slice(SequencePosition start, long length)
             int currentLength = sliceStartSegment.Memory.Length - (int)sliceStartIndex;
 
             // Position not in startSegment
-            if (currentLength < length)
+            if (currentLength < length || currentLength == length && sliceStartSegment != endObject)
```

Comment:
Can you tell me which exact test fails currently if we don't make this change?

Reply:
I already wrote about this. `if sliceStartSegment != endObject` then if we want return the same result as other methods then we should return sequence with next segment.
If we can return different results that this optimization can not changed.